Plugins/Public/event/Main.cpp

```diff
@@ -658,7 +685,50 @@ void __stdcall CharacterSelect_AFTER(struct CHARACTER_ID const & cId, unsigned i
 	}
 }
 
-void __stdcall GFGoodBuy_AFTER(struct SGFGoodBuyInfo const &gbi, unsigned int iClientID)
+void TradeEventNotice(uint iClientID, map<string, TRADE_EVENT>::iterator iter)
+{
+	Universe::IBase* base = Universe::get_base(iter->second.uEndBase);
+	const Universe::ISystem* sys = Universe::get_system(base->iSystemID);
+	const GoodInfo* gi = GoodList::find_by_id(iter->second.uCommodityID);
+	if (iter->second.eventDescription.empty())
+		iter->second.eventDescription = "None";
+
+	wstring pagetext =
+		L"<TRA bold=\"true\"/><TEXT>You've entered the " + stows(iter->second.sEventName) + L" Event</TEXT><TRA bold = \"false\"/><PARA /><PARA />"
+
+		L"<TRA bold=\"true\"/><TEXT>Mission Objective:</TEXT><TRA bold=\"false\"/><PARA/>"
+		L"<TEXT>Transport " + HkGetWStringFromIDS(gi->iIDSName) + L" to " + HkGetWStringFromIDS(base->iBaseIDS) + L" in the " + HkGetWStringFromIDS(sys->strid_name) + " system to complete this mission.</TEXT><PARA /><PARA />"
+
+		L"<TRA bold=\"true\"/><TEXT>Reward Bonus:</TEXT><TRA bold=\"false\"/><PARA/>"
+		L"<TEXT>$" + stows(to_string(iter->second.iBonusCash)) + " per unit</TEXT><PARA /><PARA />"
+
+		L"<TRA bold=\"true\"/><TEXT>Ship Class:</TEXT><TRA bold=\"false\"/><PARA/>"
+		L"<TEXT>Restricted to: " + iter->second.allowedShipClass.c_str() + "</TEXT><PARA /><PARA />"
+
+		L"<TRA bold=\"true\"/><TEXT>Description:</TEXT><TRA bold=\"false\"/><PARA/>"
+		L"<TEXT>" + stows(iter->second.eventDescription) + L"</TEXT><PARA /><PARA />";
+
+	wchar_t titleBuf[4000];
+	_snwprintf(titleBuf, sizeof(titleBuf), L"Trade Event Information");
+
+	wchar_t buf[4000];
+	_snwprintf(buf, sizeof(buf), L"<RDL><PUSH/>%ls<POP/></RDL>", pagetext.c_str());
+
+	HkChangeIDSString(iClientID, 500000, titleBuf);
```

Comment:
Those two IDSes are already in use. There is an infocard in place reserved for FLHook stuff already.
Replace with 267144 and 267145 please.